### ISSUE TITLE
Install/Compile Error RcppParallel - Linux CentOS

### ISSUE THREAD

**Issue by mthebie99** [RcppCore/RcppParallel#125]
I am unable to install latest RcppParallel for R 3.6.1 under Linux.  The error messages are not unlike some I've seen in other issues related to tbb. The architecture is being picked up so its not associated with issues suggesting modifications to the Makefile (tbs_os, etc). I get the SAME results using various install methods as well including:

1) install.packages(pkgs="RcppParallel",lib="/hb/software/apps/R/gnu-3.6.1Bio/lib64/R/library").  --- Installs from CRAN
2) Download from git latest then,
R CMD build RcppParallel
R CMD INSTALL RcppParallel_5.0.2-9000.tar.gz
3) Same as above only after running "gmake" on tbb to create architecture specific tbb lib

I tried using GCC 7.3.0 and GCC 5.4.0. Same results.
I have not tried a different version of R since the applications cannot support >3

This install is for an HPC Cluster.

**Results:**

WARNING: ignoring environment value of R_HOME
* installing *source* package ‘RcppParallel’ ...
** package ‘RcppParallel’ successfully unpacked and MD5 sums checked
** using staged installation
WARNING: ignoring environment value of R_HOME
** preparing to configure package 'RcppParallel' ...
*** configured file: 'src/Makevars.in' => 'src/Makevars'
** finished configure for package 'RcppParallel'
Warning messages:
1: In gsub(needle, replacement, contents, fixed = TRUE) :
  argument 'replacement' has length > 1 and only the first element will be used
2: In gsub(needle, replacement, contents, fixed = TRUE) :
  argument 'replacement' has length > 1 and only the first element will be used
** libs
Created ../build/lib_release directory
../../build/Makefile.tbb:32: CONFIG: cfg=release arch=x86_64 compiler=gcc target=linux runtime=cc5.4.0_libc2.17_kernel3.10.0
g++: error: WARNING:: No such file or directory
g++: error: ignoring: No such file or directory
g++: error: environment: No such file or directory
g++: error: value: No such file or directory
g++: error: of: No such file or directory
g++: error: R_HOME: No such file or directory
g++: error: WARNING:: No such file or directory
g++: error: ignoring: No such file or directory
g++: error: environment: No such file or directory
g++: error: value: No such file or directory
g++: error: of: No such file or directory
g++: error: R_HOME: No such file or directory
make[2]: *** [concurrent_hash_map.o] Error 1
make[1]: *** [tbb_release] Error 2
cp: cannot stat ‘tbb/build/lib_release/libtbb*.*’: No such file or directory
make: *** [tbb] Error 1
ERROR: compilation failed for package ‘RcppParallel’
* removing ‘/hb/software/apps/R/gnu-3.6.1Bio/lib64/R/library/RcppParallel’

The downloaded source packages are in
	‘/tmp/RtmpQNPioL/downloaded_packages’
Updating HTML index of packages in '.Library'
Making 'packages.html' ... done

**Host /GCC Environment**

OS: CentOS Linux release 7.7.1908 (Core)

There are system installs of TBB, but from my understanding of packaging these are not looked at (only the git/cran tbb is used for the build). But just incase this is relevant, here is the system install version:     
tbb.x86_64                           4.1-9.20130314.el7         @base           
tbb-devel.x86_64                     4.1-9.20130314.el7         @base     

gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/opt/ohpc/pub/compiler/gcc/5.4.0/libexec/gcc/x86_64-unknown-linux-gnu/5.4.0/lto-wrapper
Target: x86_64-unknown-linux-gnu
Configured with: ../configure --disable-multilib --enable-languages=c,c++,fortran --prefix=/opt/ohpc/pub/compiler/gcc/5.4.0
Thread model: posix
gcc version 5.4.0 (GCC) 

**R environment **
[R-Sysenv.pdf](https://github.com/RcppCore/RcppParallel/files/4974195/R-Sysenv.pdf)

** Current installed R packages**
[R-Pkgs.pdf](https://github.com/RcppCore/RcppParallel/files/4974199/R-Pkgs.pdf)

**Comment by kevinushey** [comment]
Can you share the contents of your R Makeconf? That is, the file located at `$(R RHOME)/etc/Makeconf`.

**Comment by kevinushey** [comment]
> g++: error: WARNING:: No such file or directory

I wonder if the configured compiler paths need to use a full path to the compiler, rather than using whatever happens to be found on the PATH. (Or maybe the PATH isn't set appropriately for when the installation happens?)

**Comment by mthebie99** [comment]
No Makeconf is in the bundle download.  But here is what is available:
find . -name "Make*"
./src/Makevars.in
./src/tbb/Makefile
./src/tbb/build/Makefile.tbbproxy
./src/tbb/build/Makefile.tbbmalloc
./src/tbb/build/Makefile.rml
./src/tbb/build/Makefile.tbb
./src/tbb/build/Makefile.test
./src/tbb/src/Makefile
./src/Makevars

**Comment by mthebie99** [comment]
Wait - sorry. I was looking in the download.
Here is the one under R_HOME (lib64)
[RHOME-Makeconf.pdf](https://github.com/RcppCore/RcppParallel/files/4974315/RHOME-Makeconf.pdf)

**Comment by mthebie99** [comment]
@kevinushey  Sorry for delay in reply to your question on PATH (to gcc). The output of R syspath() shows that the path to GCC is there and precedes any calls to a GCC which may be installed to the OS System (/usr/bin/gcc). Where "xxx, yyy,zzz are non-relevant [to this issue] preceding paths", and the first entry is the location of GCC.
PATH: "xxx, yyy, zzz, /opt/ohpc/pub/compiler/gcc/5.4.0/bin:/hb/software/admin/bin:/opt/ohpc/pub/utils/prun/1.3:/opt/ohpc/pub/utils/autotools/bin:/opt/ohpc/pub/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin"

So the only next possibility is to change the Makeconf (from what you are saying), but I don't know the appropriate place or line item to update.  Can you provide further guidance as to this suggestion to try?  Thank. you for your help 👍

**Comment by kevinushey** [comment]
Sorry, looking at this again I think the issue might be the fact that `R_HOME` is set and this makes R unhappy. See both of:

```
WARNING: ignoring environment value of R_HOME
```

and

```
g++: error: WARNING:: No such file or directory
g++: error: ignoring: No such file or directory
g++: error: environment: No such file or directory
g++: error: value: No such file or directory
g++: error: of: No such file or directory
g++: error: R_HOME: No such file or directory
g++: error: WARNING:: No such file or directory
g++: error: ignoring: No such file or directory
g++: error: environment: No such file or directory
g++: error: value: No such file or directory
g++: error: of: No such file or directory
g++: error: R_HOME: No such file or directory
```

I think the issue might ultimately be that `R_HOME` is being set (as an environment variable) somewhere, and this is causing issues for R. Normally, `R_HOME` should be left unset (this is set by R itself when it is initialized.)

**Comment by mthebie99** [comment]
Kevin,
I think you will find the solution I stumbled on "interesting".
The R_HOME was not assigned outside of the R initialization.
However, the variable "RHOME" was set outside of the R initialization.
Removing this exported variable assignment corrected the error condition. 
I have no idea why, but it worked... Thanks for the "hints" - a 3rd eye is ALWAYS helpful and I appreciated your time in taking a look at this. 
Issue resolved and RcppParallel is installed.
